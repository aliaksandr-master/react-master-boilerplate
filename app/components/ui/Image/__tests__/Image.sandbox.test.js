/* eslint-env browser, jest */


import React from 'react';
import renderer from 'react-test-renderer';
import { wrapSandbox } from '../../../../vendor/react-sandbox/sandbox-staging';
import sandboxImage from '../Image.sandbox';
import Test from '../../../test/Test';



describe('rendering', () => {
  wrapSandbox(sandboxImage).forEach(({ name, template, props }) => {
    it(`should render sandbox "${name}"`, () => {
      const Image = template();
      const component = renderer.create(
        <Test>
          <Image {...props} />
        </Test>
      );

      const tree = component.toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
