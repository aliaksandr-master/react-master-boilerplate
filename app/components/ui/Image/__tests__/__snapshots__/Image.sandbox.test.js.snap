// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering should render sandbox "default" 1`] = `
<div>
  <div
    className="Image Image--freeTransform"
  >
    <div
      className="Image__in"
      style={
        Object {
          "paddingBottom": "0%",
        }
      }
    >
      <img
        className="Image__img"
        src="/apple-icon-precomposed.png"
      />
    </div>
  </div>
</div>
`;

exports[`rendering should render sandbox "image_static" 1`] = `
<div>
  <div
    className="Image Image--freeTransform"
  >
    <div
      className="Image__in"
      style={
        Object {
          "paddingBottom": "0%",
        }
      }
    >
      <img
        className="Image__img"
        src="/apple-icon-precomposed.png"
      />
    </div>
  </div>
</div>
`;
